### ISSUE TITLE
Log for guys at the support

### ISSUE THREAD

**Issue by slowbackspace** [trezor/trezor-suite#239]
Design Show log: https://app.zeplin.io/project/5dadb7820bdfd3832e04afca/screen/5dde6f7ca6b4581f75605531

some ideas:
- Show basic info in more human readable format (fw version, bootloader?, app version, enabled/disabled features)
- log events such as device connected/disconnected, adding accounts, etc...
- log content of all reducers inside the "Log" component for easier debugging?
- place links to trezor support, trezor wiki
- basic quick fix? (clearing cache, storage, trying different browser)

current log from the next wallet:
<img width="795" alt="Screenshot 2019-12-03 at 10 50 53" src="https://user-images.githubusercontent.com/6961901/70039789-c87da800-15ba-11ea-9806-4ee6015c2bd5.png">

**Comment by ZdenekSL** [comment]
Update log to be better readable by humans.

**Comment by glacierbits** [comment]
First version of this feature will only have 2 buttons (copy to clipboard + export log) and a toggle to include/exclude xPUBs. Proposal is to appear in Zeplin today. 

Thorough implementation of our support team (as Freshdesk widget or custom solution) into the Suite interface needs to assessed and specified first (WIP).

**Comment by glacierbits** [comment]
^^ https://app.zeplin.io/project/5dadb7820bdfd3832e04afca/screen/5dde6f7ca6b4581f75605531 (cc @ZdenekSL )

**Comment by ZdenekSL** [comment]
@vladimirvolek

**Comment by yohaaan** [comment]
Support notes (previously discussed with @TomasZorvan and @matejzak )

**What additional info would be useful for support**
- used browser info - currently used browser - clearly visible
- differentiate the type of entry by color (same as in inspect tool - assuming that the generated file would be capable of keeping the color)
- export log feature (to prevent the necessity of copy+pasting into a text file)
- info on FW version from which the update was made (goal: to see if multiple FW version were skipped previously)
- **idea**: optimize the log for some 3rd party tool - to improve "readability" for average user/non-tech reader + select the given tool **or** directly improve the readability of the log
- **idea**: Errors messages could be rephrased - more explanatory - example: Invalid passphrase - it's not invalid, just different from the one that was entered previously, misleading

**Comment by tsusanka** [comment]
Note: we could put this also in the "create github issue" button (#2994), that would help debugging tremendously. Of course some warning what the log contains would have to be present.

**Comment by tsusanka** [comment]
We currently have this:

![image](https://user-images.githubusercontent.com/1835345/104917165-5705e200-5993-11eb-8697-388d02e4ae3d.png)

---

As per @yohaaan's suggestions:

> used browser info - currently used browser - clearly visible

Present.

> differentiate the type of entry by color (same as in inspect tool - assuming that the generated file would be capable of keeping the color)

We would have to export the file into HTML and that seems like a too much effort. So if you survive I would not do it.

> export log feature (to prevent the necessity of copy+pasting into a text file)

Present.

![image](https://user-images.githubusercontent.com/1835345/104918051-7fdaa700-5994-11eb-8b92-f73d33a5d00b.png)

> info on FW version from which the update was made (goal: to see if multiple FW version were skipped previously)

That would be nice. Unfortunately we do not have that information all the time only during the update. Or we could store it in Suite's internal storage. I will open a new issue for that.

> idea: optimize the log for some 3rd party tool - to improve "readability" for average user/non-tech reader + select the given tool or directly improve the readability of the log

Since it is a simple JSON and there are a lot of [JSON](https://jsonformatter-online.com/) [formatters](https://jsoneditoronline.org) I dare to say it is out of scope.

> idea: Errors messages could be rephrased - more explanatory - example: Invalid passphrase - it's not invalid, just different from the one that was entered previously, misleading

That's a more general issue not related to the log, we should have a more expressive notifications and I think we do already.

---

Sounds OK @yohaaan?

**Comment by slowbackspace** [comment]
Heyyy, I have another idea for the log. 
After 2 years I still find logging events completely useless. More often I only want to see snapshot of app's internal state and from that I can finally figure out what is happening. I don't have enough mental capacity to replay 3000 logged events and derive the state from them. What's more important so far there wasn't single case where we actually cared for all dispatched events. There are tons of them, we cannot log them all, and chances are events you are interested in are not logged anyway.

Last time I was remotely debugging Bach's issue on his own HW I found very useful that we can just dump app state. I think the log should contain exactly this, dump of data stored inside the reducers. It is fairly trivial to implement (as a bonus should be simpler than current solution)

Fun fact: there is a bug (for more than a year) causing only that only first 199 entries are logged, then it acts like a stack and each new entry is replacing the old one.
